### DISCUSSION TITLE
Random Number Generator / Filter to support sampling

### DISCUSSION THREAD

**Discussion by mp3monster** [fluent/fluent-bit#8344] [Ideas]
When handling Traces at volume, it isn;t unheard of for organizations to reduce the number of possible traces being handled by applying a sampling strategy - https://opentelemetry.io/docs/concepts/sampling/. By having a function that generates a random number in a defined range e.g. 1-100 it would be then possible to define a filter that will drop a proportion of traces rather than labouring the backend.

What would be even more elegant would be to have a Filter that uses the function and takes a parameter defining the percentage sample to allow to continue.

**Comment by patrick-stephens** [comment]
Similar to the throttle filter I guess but sounds like a reasonable thing to do, even as a direct processor or config option on OTEL inputs for traces
